app/controllers/comments_controller.rb

```diff
@@ -18,6 +19,15 @@ def create
     end
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    @comment.destroy
+    @post = Post.find(@comment.post_id)
+
+    @post.update(comments_counter: @post.comments_counter - 1)
+    redirect_to user_post_path(current_user.id, @post.id), notice: 'Comment deleted.'
+  end
```

Comment:
- [OPTIONAL] Instead of using this logic here, you can use the after_destroy callback method in the model file for the Comment model. This will automatically do it. 👍 😎 

```ruby
after_destroy :decrement_comments_counter

private

def decrement_comments_counter
  post.decrement!(:comments_counter)
end
```